pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/DefaultCommitterRealtimeIntegrationTest.java

```diff
@@ -0,0 +1,204 @@
+ */
+public class DefaultCommitterRealtimeIntegrationTest extends RealtimeClusterIntegrationTest {
+  private File _indexDir;
+  private File _realtimeSegmentUntarred;
+
+  private static final String METADATA_FILE_NAME = "metadata.properties";
+  private static final String TARGZ_SUFFIX = ".tar.gz";
+  private static final long END_OFFSET = 500L;
+  private static final String CONSUMER_DIRECTORY = "/tmp/consumer-test";
+  private static final Logger LOGGER = LoggerFactory.getLogger(DefaultCommitterRealtimeIntegrationTest.class);
+
+  @BeforeClass
+  @Override
+  public void setUp()
+      throws Exception {
+    // Remove the consumer directory
+    File consumerDirectory = new File(CONSUMER_DIRECTORY);
+    if (consumerDirectory.exists()) {
+      FileUtils.deleteDirectory(consumerDirectory);
+    }
+
+    startZk();
+    startController();
+    addFakeBrokerInstancesToAutoJoinHelixCluster(1, true);
+    addFakeServerInstancesToAutoJoinHelixCluster(1, true);
+
+    ControllerLeaderLocator.create(_helixManager);
+
+    // Start Kafka
+    startKafka();
+    List<File> avroFiles = unpackAvroData(_tempDir);
+
+    setUpRealtimeTable(avroFiles.get(0));
+    buildSegment();
+  }
+
+  @Override
+  protected boolean useLlc() {
+    return true;
+  }
+
+  @Test
+  public void testDefaultCommitter() throws Exception {
+    IndexLoadingConfig indexLoadingConfig = new IndexLoadingConfig();
+    ServerMetrics serverMetrics = new ServerMetrics(new MetricsRegistry());
+    ServerSegmentCompletionProtocolHandler
+        protocolHandler = new ServerSegmentCompletionProtocolHandler(serverMetrics, getTableName());
+
+    SegmentCompletionProtocol.Response prevResponse = new SegmentCompletionProtocol.Response();
+    LLRealtimeSegmentDataManager.SegmentBuildDescriptor segmentBuildDescriptor = mock(
+        LLRealtimeSegmentDataManager.SegmentBuildDescriptor.class);
+
+    RealtimeSegmentZKMetadata metadata = _helixResourceManager.getRealtimeSegmentMetadata(getTableName()).get(0);
+
+    String instanceId = _helixResourceManager.getAllInstances().get(0);
+
+    SegmentCompletionProtocol.Request.Params params = new SegmentCompletionProtocol.Request.Params();
+    params.withSegmentName(metadata.getSegmentName()).withInstanceId(instanceId).withOffset(END_OFFSET);
+
+    _realtimeSegmentUntarred = new File(_indexDir.getParentFile(), metadata.getSegmentName());
+    FileUtils.copyDirectory(_indexDir, _realtimeSegmentUntarred);
+
+    TarGzCompressionUtils.createTarGzOfDirectory(_realtimeSegmentUntarred.getAbsolutePath());
+
+    when(segmentBuildDescriptor.getSegmentTarFilePath()).thenReturn(_realtimeSegmentUntarred + TARGZ_SUFFIX);
+    when(segmentBuildDescriptor.getBuildTimeMillis()).thenReturn(0L);
+    Map<String, File> metadataPaths = new HashMap();
+    metadataPaths.put(metadata.getSegmentName(), new File(_indexDir, METADATA_FILE_NAME));
+
+    when(segmentBuildDescriptor.getMetadataFiles()).thenReturn(metadataPaths);
```

Comment:
since you are testing default commit in this test, you can omit the getMetadtaFiles mock